### PR TITLE
Support modifying Client headers

### PR DESCRIPTION
**Pull request by bergundy**
Does not address need to set per-call headers, make the Client headers mutable so they can be updated from Lang
Related: #303

### CODE REVIEW

File: client/src/lib.rs [hunk 1]
```diff
@@ -80,9 +80,9 @@ pub struct ClientOptions {
     /// in all RPC calls. The server decides if the client is supported based on this.
     pub client_version: String,
 
-    /// Other non-required static headers which should be attached to every request
+    /// Other non-required headers which should be attached to every request
     #[builder(default)]
-    pub static_headers: HashMap<String, String>,
+    pub headers: Arc<Mutex<HashMap<String, String>>>,
```
Comment (Sushisource) [inline client/src/lib.rs:85]: Use a parking_lot mutex to avoid fallible locking

Comment (Sushisource) [reply client/src/lib.rs:85]: I also don't really love the idea of this being what gets mutated. IMO the options should remain static.

We could change this to be `default_headers` and then put this as a field on the interceptor specifically named `header_overrides` and that can be changed with a function call.

Comment (bergundy) [reply client/src/lib.rs:86]: Can you explain why that's better?

File: client/src/lib.rs [hunk 2]
```diff
@@ -394,6 +408,7 @@ pub struct WorkflowTaskCompletion {
 #[derive(Clone)]
 pub struct ServiceCallInterceptor {
     opts: ClientOptions,
+    headers: Arc<Mutex<HashMap<String, String>>>,
```
Comment (Sushisource) [inline client/src/lib.rs:411]: Should probably doc here that they are only ever meant to be _read_ from here, not written.

Additionally, probably makes a lot of sense for this to be a `RwLock`

**Review by Sushisource — APPROVED** [review]

**Pull request merged**